### ISSUE TITLE
Automatic download feature integration

### ISSUE THREAD

**Issue by bjester** [learningequality/kolibri#10390]
Depends on: https://github.com/learningequality/kolibri/pull/10365 and https://github.com/learningequality/kolibri/issues/10388

## Background
A new 0.16 device setting `enable_automatic_download` controls the automatic content import implemented in https://github.com/learningequality/kolibri/pull/10365. The setting defaults to `True`, which means this setting will be enabled upon upgrading from prior versions. The automatic import relies on a facility sync to identify lessons and exams that changed as a result of the sync, in order to create solely the necessary request models that track the content that needs downloaded or removed. Since upgrading Kolibri would break this arrangement, a new version upgrade migration is needed to perform this request model creation for the entire database. In addition, we can consolidate this functionality into a utility that can be executed when the user toggles this device setting from disabled to enabled.

Admin imported content is not automatically removed. The creation of lessons and exams also requires resources to be imported. An edge case could exist where a facility requires materials for certain assignments, and the remixing of those resources into new assignments could result in removal of the content for the newly assigned items if the previously assigned items were synced to the device.

### A new task
a) It should call `synchronize_content_requests` for all facilities / datasets on the device
b) It should enqueue the `automatic_resource_import` task afterwards

### The `DeviceSettings` option `enable_automatic_download`
a) It should enqueue the above task when the setting `enable_automatic_download` is toggled from disabled to enabled

### The function [`synchronize_content_requests`](https://github.com/learningequality/kolibri/blob/c588e4b5d73ae0342d3ff56dd2f4f2bf400e698b/kolibri/core/content/utils/content_request.py#L49)
a) It should be refactored into 2 more functions, so that the loop logic for creation of download and removal requests can be separately invoked with individual assignments

### The class `ContentAssignmentManager`
a) It should have new classmethods `on_downloadable_assignment` and `on_removable_assignment` which accept a callable each
b) Its classmethods `on_downloadable_assignment` and `on_removable_assignment` should connect separate callables to the save hook of every model the manager is attached to, and execute the input callable with any new downloadable/removable assignments that matches its filters (if defined) when its callables are invoked

### New signal functions `add_download_requests` and `add_removal_requests`
a) They should exist in `content/signals.py` 
b) They should accept a list of assignments as computed in (b) above
c) They should invoke the individual functions refactored from `synchronize_content_requests` above, only when when `get_device_setting("enable_automatic_download", default=False)` is `True`

### The class [`KolibriContentConfig`](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/content/apps.py#L13)
a) It should call the `ContentAssignmentManager` classmethods `on_downloadable_assignment` and `on_removable_assignment` with the individual signal callables `add_download_requests` and `add_removal_requests` attaching them for signal handling
